jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletHandler.java

```diff
@@ -434,41 +437,42 @@ public void doScope(String target, Request baseRequest, HttpServletRequest reque
         // Get the base requests
         final String old_servlet_path = baseRequest.getServletPath();
         final String old_path_info = baseRequest.getPathInfo();
-        final PathSpec old_path_spec = baseRequest.getPathSpec();
+        final HttpServletMapping old_http_servlet_mapping = baseRequest.getHttpServletMapping();
 
         DispatcherType type = baseRequest.getDispatcherType();
 
         ServletHolder servletHolder = null;
         UserIdentity.Scope oldScope = null;
 
-        MappedResource<ServletHolder> mapping = getMappedServlet(target);
+        MappedServlet mapping = getMappedServlet(target);
         if (mapping != null)
         {
             servletHolder = mapping.getResource();
 
-            if (mapping.getPathSpec() != null)
+            PathSpec pathSpec = mapping.getPathSpec();
+            if (pathSpec != null)
             {
-                PathSpec pathSpec = mapping.getPathSpec();
                 String servletPath = pathSpec.getPathMatch(target);
                 String pathInfo = pathSpec.getPathInfo(target);
 
+                HttpServletMapping httpServletMapping = mapping.getHttpServletMapping(servletPath);
```

Comment:
Just to double down on my comments.   The getHttpServletMapping internally does pretty much all the work that is done in getPathMatch and get PathInfo above.   It just needs to remember the results and return them in the JettyHttpServletMapping and we can do only 1 slice&dice rather than 3.